src/run.c

```diff
@@ -225,6 +231,64 @@ Port inject_bytes(Net* net, Bytes *bytes) {
   return port;
 }
 
+/// Returns a λ-Encoded Ctr for a RESULT_OK: λt ((t RESULT_OK) val)
+Port inject_ok(Net* net, Port val) {
+  if (!get_resources(net, tm[0], 0, 3, 1)) {
+    fprintf(stderr, "inject_ok: failed to get resources\n");
+    return new_port(ERA, 0);
+  }
+
+  u32 v1 = tm[0]->vloc[0];
+
+  u32 n1 = tm[0]->nloc[0];
+  u32 n2 = tm[0]->nloc[1];
+  u32 n3 = tm[0]->nloc[2];
+
+  vars_create(net, v1, NONE);
+  Port var = new_port(VAR, v1);
```

Comment:
What happens if the error value is not a var, does it just cause a link interaction?

Reply:
The var `v1` is used to link two aux ports. The returned net is:
```
((RESULT_ERR (err x)) x)
```
where `RESULT_ERR` is the numeric tag of the `Result/Err` variant, and the `x` var in the snippet above is `v1` in the code.